### DISCUSSION TITLE
only working in local network?

### DISCUSSION THREAD

**Discussion by linyaohui** [Ylianst/MeshCentral#6192] [General]
dear  i install the client on some computer,only the ones located in local office is working
the ones outsite the office not working
![image](https://github.com/Ylianst/MeshCentral/assets/7878171/31dca165-781e-42d1-8573-7ae05eb91c90)

**Comment by linyaohui** [comment]
it's possible to show all the pc which installed the client ?  wlan or lan

**Comment by linyaohui** [comment]
why server url is local???
![image](https://github.com/Ylianst/MeshCentral/assets/7878171/7b6809db-51c6-49b1-b0c0-f5db9507c4df)

**Comment by si458** [comment]
**you havent followed the bug template report with has missing information we need to help you**
**and this is not a bug but a question for the community to help you**
you need to share with us your config.json file.
as when you first start meshcentral it will run in LAN mode, which means it will only work with devices in its local area network
hense why you are seeing it say 'local' in the meshagent

the is some docs here to help you get started/look through
https://ylianst.github.io/MeshCentral/meshcentral/

the is also some youtube videos here to watch https://www.youtube.com/@MeshCentral/videos
and the video you are after is lan/wan/hybrid mode - https://www.youtube.com/watch?v=gx5Fh3pQOns

**Comment by linyaohui** [comment]
hi my config.json
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "_cert": "183.6.5.44",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_port": 8853,
    "_aliasPort": 8853,
    "_redirPort": 8851,
    "_redirAliasPort": 8851
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```

**Reply by linyaohui** [reply]
my server dont have cert now,no https  only http

**Reply by si458** [reply]
to enable WAN mode so external devices can access it, you need to set `_cert` to `cert` and then set a DNS name in the value

**DONT USE AN IP ADDRESS AS IP ADDRESSES CAN CHANGE ALWAYS USE DNS!**

also dont forget to portforward your port numbers throught your routers firewall is you are using it internally and want external devices to see it

**Comment by linyaohui** [comment]
how to  enable WAN mode ?  i already      "_WANonly": true,

**Reply by si458** [reply]
i have just told you above - https://github.com/Ylianst/MeshCentral/discussions/6192#discussioncomment-9815351
to enable WAN mode so external devices can access it, you need to set `_cert` to `cert` and then set a DNS name in the value
anything in the config.json that has an underscore `_` infront of it is ignored, so you must remove the `_` for it to read the value

**Comment by linyaohui** [comment]
![image](https://github.com/Ylianst/MeshCentral/assets/7878171/6d4f0841-43e6-472b-b573-557a7e5974f2)

**Comment by linyaohui** [comment]
thank you very very much  working now
woooooooo
![Uploading image.png…]()

**Comment by linyaohui** [comment]
really good。。。。。。。。。。。

**Comment by silversword411** [comment]
Don't forget to support for the support provided 
https://github.com/si458
